src/Formats/SchemaInferenceUtils.cpp

```diff
@@ -1645,6 +1645,9 @@ DataTypePtr makeNullableRecursively(DataTypePtr type)
         return std::make_shared<DataTypeObjectDeprecated>(object_type->getSchemaFormat(), true);
     }
 
+    if (which.isObject())
```

Comment:
We don't infer new JSON right now, but use `makeNullableRecursively` during JSON data type parsing and it would make nested `Array(JSON)` to be `Array(Nullable(JSON))` and I don't want it by default. Maybe it's better to make it under setting, will do it